src/split-pane/index.ts

```diff
@@ -55,6 +56,23 @@ export class SplitPaneBase<P extends SplitPaneProperties = SplitPaneProperties>
 	private _lastSize?: number;
 	private _position = 0;
 	private _collapsed = false;
+	private _resizeResultOverridden = false;
+	private _width = 0;
+
+	@diffProperty('collapseWidth')
+	@diffProperty('direction')
+	private collapseWidthDiff(oldProperties: SplitPaneProperties, { collapseWidth, direction, onCollapse }: SplitPaneProperties) {
+		if (collapseWidth) {
+			if (direction === Direction.row || collapseWidth < this._width && this._collapsed) {
+				this._collapsed = false;
+				this._resizeResultOverridden = true;
```

Comment:
Should `onCollapse(false)` be getting called in this case? It looks like it would have been under the old logic.

Reply:
good eye